### PR TITLE
passing defined repositories to scalafmt (#28)

### PR DESCRIPTION
**Pull request by slivkamiro**

### CODE REVIEW

File: src/main/groovy/cz/alenkacz/gradle/scalafmt/ScalafmtFormatBase.groovy [hunk 1]
```diff
@@ -14,7 +15,13 @@ class ScalafmtFormatBase extends DefaultTask {
     def formatter = Scalafmt.create(cl)
             .withRespectVersion(false)
             .withDefaultVersion("1.5.1")
+            .withMavenRepositories(repositories())
 
+    private String[] repositories() {
+        project.getRepositories().stream().map { repository ->
```
Comment (augi) [inline src/main/groovy/cz/alenkacz/gradle/scalafmt/ScalafmtFormatBase.groovy]: Are you sure that the `project` field is already set at this stage? (when initializing a field)

**Review by augi — CHANGES_REQUESTED** [review]

File: src/main/groovy/cz/alenkacz/gradle/scalafmt/ScalafmtFormatBase.groovy (hunk 1 above)
Comment (slivkamiro) [reply src/main/groovy/cz/alenkacz/gradle/scalafmt/ScalafmtFormatBase.groovy]: Thanks, I'm new to gradle. Should I move the initialization of scalafmt to the run method then? Or do you have something else on your mind?

Comment (tomasherman) [reply src/main/groovy/cz/alenkacz/gradle/scalafmt/ScalafmtFormatBase.groovy]: i think moving this to method invocation is much safer, good point augi

Comment (augi) [reply src/main/groovy/cz/alenkacz/gradle/scalafmt/ScalafmtFormatBase.groovy]: I don't exactly know that the builder exactly does, but creating them in a method (that is called when needed) seems to be OK for me 👍

**Review by augi — APPROVED** [review]

**Comment by tomasherman** [comment]
so can we merge this? mr. @augi ?

**Review by augi — APPROVED** [review]

**Comment by tomasherman** [comment]
is this going to get released aswell?

**Comment by augi** [comment]
It has been already released as `1.10.0`.

**Pull request merged**